src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/entities/ReportedAdjudication.kt

```diff
@@ -37,9 +38,38 @@ data class ReportedAdjudication(
   var statusReason: String? = null,
   @Length(max = 1000)
   var statusDetails: String? = null,
-) : BaseEntity()
+) : BaseEntity() {
+  fun transition(to: ReportedAdjudicationStatus, statusReason: String? = null, statusDetails: String? = null) {
+    if (this.status.canTransitionTo(to)) {
+      this.status = to
+      this.statusReason = statusReason
+      this.statusDetails = statusDetails
+    } else {
+      throw IllegalStateException("ReportedAdjudication ${this.reportNumber} cannot transition from ${this.status} to $to")
+    }
+  }
+}
 
 enum class ReportedAdjudicationStatus {
-  ACCEPTED, REJECTED, AWAITING_REVIEW, RETURNED;
-  fun isFinal(): Boolean = this in listOf(ACCEPTED, REJECTED)
+  ACCEPTED,
+  REJECTED,
+  AWAITING_REVIEW {
+    override fun nextStates(): List<ReportedAdjudicationStatus> {
+      return listOf(ACCEPTED, REJECTED, RETURNED, AWAITING_REVIEW)
```

Comment:
The AWAITING_REVIEW -> AWAITING_REVIEW is so that a draft can be reported by the original reporter before a review has looked at it - which is currently allowed (and I think seems right)

Reply:
Yep